.github/workflows/auto-approve.yml

```diff
@@ -0,0 +1,11 @@
+name: Auto approve
+on: pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: hmarr/auto-approve-action@v2.0.0
+      if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
```

Comment:
I think it will save time and resources if you move or duplicate this condition on the job itself. That way the job won't even run if this doesn't match.